# Lending club case study

In [1]:
# import libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
df = pd.read_csv("./data/loan.csv", low_memory=False)
df.shape

(39717, 111)

## Find nan values

In [5]:
dfnull = pd.DataFrame()
dfnull['nans'] = df.isnull().sum()
dfnull['nans'].value_counts()

39717    54
0        43
56        2
25682     1
71        1
36931     1
697       1
1075      1
50        1
38577     1
39        1
2459      1
12940     1
11        1
2         1
Name: nans, dtype: int64

### 54 columns have 39717 'nans' - same as the number of rows. Drop these columns.

In [6]:
df = df.dropna(axis=1, how='all')
df.shape

(39717, 57)

## Columns with same values - these are not useful for analysis

In [7]:
same_columns = {}
for c in df.columns:
    if df[c].value_counts().shape[0] == 1:
        same_columns[c] = list(df[c].unique())
same_columns

{'pymnt_plan': ['n'],
 'initial_list_status': ['f'],
 'collections_12_mths_ex_med': [0.0, nan],
 'policy_code': [1],
 'application_type': ['INDIVIDUAL'],
 'acc_now_delinq': [0],
 'chargeoff_within_12_mths': [0.0, nan],
 'delinq_amnt': [0],
 'tax_liens': [0.0, nan]}

### Drop columns with same values

In [8]:
df = df.drop(same_columns.keys(), axis=1)
df.shape

(39717, 48)